webapi/src/main/scala/org/knora/webapi/slice/common/api/ApiV2.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright © 2021 - 2023 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.slice.common.api
+
+import sttp.model.HeaderNames
+import sttp.model.MediaType
+import sttp.tapir.Codec
+import sttp.tapir.Codec.PlainCodec
+import sttp.tapir.CodecFormat
+import sttp.tapir.DecodeResult
+import sttp.tapir.EndpointIO
+import sttp.tapir.EndpointInput
+import sttp.tapir.Validator
+import sttp.tapir.header
+import sttp.tapir.query
+
+import dsp.errors.BadRequestException
+import org.knora.webapi.ApiV2Complex
+import org.knora.webapi.ApiV2Schema
+import org.knora.webapi.JsonLdRendering
+import org.knora.webapi.MarkupRendering
+import org.knora.webapi.Rendering
+import org.knora.webapi.messages.util.rdf.JsonLD
+import org.knora.webapi.messages.util.rdf.RdfFormat
+import org.knora.webapi.slice.common.api.KnoraResponseRenderer.FormatOptions
+
+object ApiV2 {
+
+  /**
+   * The names of the supported Http Headers.
+   */
+  object Headers {
+
+    /**
+     * The name of the HTTP header that can be used to specify how markup should be returned with
+     * text values.
+     */
+    val xKnoraAcceptMarkup: String = "x-knora-accept-markup"
+
+    /**
+     * The name of the HTTP header in which results from a project can be requested.
+     */
+    val xKnoraAcceptProject: String = "x-knora-accept-project"
+
+    /**
+     * The name of the HTTP header in which an ontology schema can be requested.
+     */
+    val xKnoraAcceptSchemaHeader: String = "x-knora-accept-schema"
+
+    /**
+     * The name of the HTTP header that can be used to request hierarchical or flat JSON-LD.
+     */
+    val xKnoraJsonLdRendering: String = "x-knora-json-ld-rendering"
+  }
+
+  object QueryParams {
+
+    /**
+     * The name of the URL parameter in which an ontology schema can be requested.
+     */
+    val schema: String = "schema"
+
+    /**
+     * The name of the URL parameter that can be used to specify how markup should be returned
+     * with text values.
+     */
+    val markup: String = "markup"
+  }
+
+  object Inputs {
+    import Codecs.{apiV2SchemaListCodec, jsonLdRenderingListCodec, markupRenderingListCode}
+
+    // ApiV2Schema inputs
+    val defaultApiV2Schema: ApiV2Schema = ApiV2Complex
+    private val apiV2SchemaHeader = header[Option[ApiV2Schema]](Headers.xKnoraAcceptSchemaHeader)
```

Comment:
Some of the `val` names in this file could be simplified, because they are already in the `ApiV2` object.

Reply:
Not sure what you are referring to? Can you give an example please?

Reply:
For example `defaultApiV2Schema` could be reduced to `defaultSchema` and `apiV2SchemaHeader` to `schemaHeader`. At the end it is nothing bad, but just less to read.